### PR TITLE
Add CMake support

### PR DESCRIPTION
**Pull request by rajkan01**
- Added .travis.yml config to pick this example for Travis CI build
- Update mbed-os.lib to pick always the latest master

@hugueskamba @0xc0170

### CODE REVIEW

File: mbed_app.json [hunk 1]
```diff
@@ -1,8 +1,6 @@
 {
-    "requires": ["bare-metal"],
     "target_overrides": {
       "*": {
-        "target.c_lib": "small",
```
Comment (hugueskamba) [inline mbed_app.json]: Why removing this?

Comment (rajkan01) [reply mbed_app.json]: I just now realized and reverting these changes

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2020 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.18.2 FATAL_ERROR)
```
Comment (hugueskamba) [inline CMakeLists.txt]:
```suggestion
cmake_minimum_required(VERSION 3.19.0 FATAL_ERROR)
```
The latest version has required fixes for Arm Compiler 6 support.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        - popd
+        - arm-none-eabi-gcc --version
+        - pip install --upgrade cmake
+        - pip install --upgrade mbed-tools
+        # Fetch mbed-os: We use manual clone, with depth=1 and --single-branch to save time.
+        - >-
+          git clone --depth=1 --single-branch https://github.com/ARMmbed/mbed-os.git;
+        - >-
+          mbedtools configure -t GCC_ARM -m ${TARGET_NAME};
+          mkdir -p build
```
Comment (hugueskamba) [inline .travis.yml]:
```suggestion
```

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      script:
+        - cd build && cmake .. -GNinja -DCMAKE_BUILD_TYPE=${PROFILE} && cmake --build .
```
Comment (hugueskamba) [inline .travis.yml]:
```suggestion
        - mbedtools build -t GCC_ARM -m ${TARGET_NAME} -b ${PROFILE}
```

**Review by hugueskamba — APPROVED** [review]

**Review by 0xc0170 — COMMENTED** [review]
Travis will fail as 3.19 is not in pypi yet, please update as we agreed (the issue IOTOSM-2617)

**Comment by 0xc0170** [comment]
> Travis will fail as 3.19 is not in pypi yet, please update as we agreed (the issue IOTOSM-2617)

Or not... as not yet enabled for this repo :-)

**Comment by 0xc0170** [comment]
Wait, to development? This should be to feature-cmake only update, not development

**Review by 0xc0170 — APPROVED** [review]

**Comment by 0xc0170** [comment]
PR to master was merged. Once CI is green, this is ready for merge

**Comment by hugueskamba** [comment]
> PR to master was merged. Once CI is green, this is ready for merge

It is not yet ready to be merged because there is an issue with the tools (https://github.com/ARMmbed/mbed-tools/issues/109).

**Comment by 0xc0170** [comment]
Ok, this is not yet enabled in our CI so not blocking it yet.

**Comment by jainvikas8** [comment]
Please update Mbed CLI 2 command line to match with 3.6.0 future release

**Comment by 0xc0170** [comment]
@rajkan01 Ready to be merged now?

**Comment by rajkan01** [comment]
> 
> 
> @rajkan01 Ready to be merged now?

Yes this PR is good to go and necessary keyword support added in latest mbed-tools release

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        # can't permit the Travis-preinstalled copy to survive, as the Travis default
+        # path lists the Travis CMake install location ahead of any place where apt
+        # would install CMake to. Instead of apt removing or upgrading to a new CMake
+        # version, we must instead delete the Travis copy of CMake.
+        - sudo rm -rf /usr/local/cmake*
+        - pip install --upgrade mbed-tools
+        - pip install prettytable==0.7.2
+        - pip install future==0.16.0
+        - pip install "Jinja2>=2.10.1,<2.11"
+        - pip install "intelhex>=1.3,<=2.2.1"
```
Comment (hugueskamba) [inline .travis.yml:80]:
```suggestion
        - pip install "intelhex>=1.3,<=2.2.1"
        # Fetch mbed-os: We use manual clone, with depth=1 and --single-branch to save time.
        - >-
          git clone --depth=1 --single-branch https://github.com/ARMmbed/mbed-os.git;
        - >-
```

**Review by hugueskamba — CHANGES_REQUESTED** [review]

**Review by 0xc0170 — APPROVED** [review]
once updated, LGTM

File: .travis.yml (hunk 5 above)
Comment (rajkan01) [reply .travis.yml:80]: @hugueskamba This is no longer required as we decided to call `mbedtools deploy` command instead git clone which I have already added under `script:`

**Review by hugueskamba — APPROVED** [review]

**Pull request merged**